.github/workflows/codecov.yml

```diff
@@ -0,0 +1,70 @@
+name: Generate and upload code coverage
+
+on:
+  #On push to common branches, this computes the "bases stats" for PRs
+  push:
+    branches:
+      - main
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  All_Tests:
+    name: All tests
+    runs-on: ubuntu-20.04
+    strategy:
+      matrix:
+        nim-options: [
+          ""
+        ]
+        test-program: [
+          "test"
+        ]
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Environment setup
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y lcov build-essential git curl
+          mkdir coverage
+
+      - name: Setup Nimble
+        uses: "./.github/actions/install_nimble"
```

Comment:
Is this something we could make into a global github action? Something like https://github.com/jiro4989/setup-nim-action

Reply:
https://github.com/nim-lang/nimble/pull/1122